Noise models input refactor

jjclavijo opened the pull request:
Terminé de eliminar la dependencia de Control.

control.py ahora es solo un parser para los archivos .ctl (de la versión anterior del programa). Pero es opcional, si el input es un .toml no pasa nada.

Lo mas valioso de control.py, ademas de ser un ejemplo de todo lo que está mal cuando programás, es que estan los chequeos de parámetros y implicitamente está ahí todo lo necesario para saber cuales son los parámetros. Después lo convertiré en documentación.

En observations.py eliminé la dependencia también, pero no borré la clase porque ahí están los writers para los archivos de salida que usa el programa original. En realidad esto no es un requisito, no lo borré porque no quería borrar sin pensar.

gonzalogacc committed:
refactor(noise-test): change test to accept a toml config string

gonzalogacc committed:
refactor(test-nose): remove control obj from test test_noise_compare_GGM

gonzalogacc committed:
refactor(noise): changes to user toml config files

gonzalogacc committed:
refactor(noise): call noise function directly without dispatcher, arguments are all named, missing arguments raise exception at the function

gonzalogacc committed:
refactor(helpers): mactrick function to helper DRY

gonzalogacc committed:
refactor(noise): refactor stuff

gonzalogacc committed:
fix(create_h): add return component to create_h_ model functions

gonzalogacc committed:
More testing fix

jjclavijo committed:
Eliminada dependencia de Clase Control

El archivo control.py ahora solo tiene un parser de los viejos archivos
.ctl. Devuelve un diccionario igual que si leyera un toml.

Por ahora lo dejamos como documentación mas que nada, porque ahí están
implícitamente registradas la mayoría de las opciones que hay que tener
en cuenta.

También se modificó observations, pero todavía no lo eliminamos porque
se requiere para que el programa pueda seguir escribiendo los archivos
de salida en el formato del programa original (esto por ahora no lo
borramos para no romper todo pero no es necesario).

Se comentó un test inutil en test_noise_creation.

Se agregó un script de bash para chequear que el programa corre sobre
los archivos ctl de valid_ctl. (muy trogloditamente).

Queda bastante trabajo con los parámetros todavía.

gonzalogacc commented:
Tenemos que agregarle una carpeta examples con un .toml adentro

gonzalogacc reviewed (changes requested):
Agreguemos una carpeta de ejemplos con un par de archivos de config .toml apra tener a mano y correr el programa

jjclavijo committed:
Independencia total de hector.

Abandonamos el formato .cli y la entrada por línea de comandos.

Conservamos en eletor.control una utilidad independiente que convierte
`.ctl` -> `.toml` y `.toml`  -> `.ctl`+`.cli`
El archivo `.cli` es porque hay parámetros que hector sólo acepta por la
línea de comandos.

Si intentas correr eletor con un archivo .ctl te avisa que no se puede.

Los archivos valid_ctls están generados con ese esquema. Hay que generar
mas.

jjclavijo commented:
Nos independizamos del formato .ctl

en [la carpeta correspondeente](/test/valid_ctls) hay dos toml que andan ahora.

[El modulo control](/src/eletor/control.py) Ahora es solo un conversor ida y vuelta entre ctl y toml.

Cuando pasas de toml a ctl se agrega un archivo .cli par apasar los parámetros que van por línea de comandos.

La ide para terminar de independizarnos de hector en los tests es:


1. Generar un `xxxx.toml` válido
2. Convertirlo con `python -m eletor.control -I xxxx.toml`
3. Correr hector con `simulatenoise -i xxxx.ctl < xxxx.cli`
4. Ver cómo comparar el archivo .mom

Para comparar la salida de hector tenemos que dejar el writer de archivos .mom
en eletor, pero el camino es eliminar todo el resto de observations.py
